web/client/components/maps/MapCard.jsx

```diff
@@ -40,15 +61,56 @@ const MapCard = React.createClass({
     },
     render: function() {
 
+        var availableAction = [{
+            onClick: () => this.props.viewerUrl(this.props.map),
+            glyph: "chevron-right",
+            tooltip: <Message msgId="manager.openInANewTab" />
+        }];
+
+        if (this.props.map.canEdit === true) {
+            availableAction.unshift({
+                 onClick: () => this.displayDeleteDialog(),
+                 glyph: "1-close",
+                 tooltip: <Message msgId="manager.deleteMap" />
+         }, {
+                 onClick: () => this.onEdit(this.props.map),
+                 glyph: "wrench",
+                 tooltip: <Message msgId="manager.editMapMetadata" />
+         });
+        }
         return (
-           <GridCard className="map-thumb" style={this.getCardStyle()} header={this.props.map.title || this.props.map.name} actions={
-                   [{
-                       onClick: () => this.props.viewerUrl(this.props.map),
-                       glyph: "chevron-right",
-                       tooltip: <Message msgId="manager.openInANewTab" />
-                   }]
-               }><div className="map-thumb-description">{this.props.map.description}</div></GridCard>
+           <GridCard className="map-thumb" style={this.getCardStyle()} header={this.props.map.title || this.props.map.name} actions={availableAction}>
+               <div className="map-thumb-description">{this.props.map.description}</div>
+               <MetadataModal ref="metadataModal" show={this.state.displayMetadataEdit} onHide={this.close} onClose={this.close} map={this.props.map} onMetadataEdit={this.props.onMetadataEdit}/>
+               <ConfirmModal ref="deleteMapModal" show={this.state.displayDeleteDialog} onHide={this.close} onClose={this.close} onConfirm={this.onConfirmDelete} confirmText={<Message msgId="manager.deleteMap" />} cancelText={<Message msgId="cancel" />} body={<Message msgId="manager.deleteMapMessage" />} />
+           </GridCard>
         );
+    },
+    close() {
+        this.setState({
+            displayMetadataEdit: false,
+            displayDeleteDialog: false
+        });
+    },
+    open() {
+        this.setState(
```

Comment:
This will be removed in my next cleaning phase, please decide if you to do it now to ease my work